### ISSUE TITLE
not downloading to local server without internet connection

### ISSUE THREAD

**Issue by ahmadhsalim** [4Q-s-r-o/ota_update#63]
Our ota update server and the android phone is in a wifi network without internet connection. The android phone is able to download the apk file via the browser. But the library is failing to download the ota update if there is no internet connection. We cannot give internet access to the mobile phone but we want the app to update. Is this possible?

```
D/FLUTTER OTA(19650): OTA UPDATE ON LISTEN
D/FLUTTER OTA(19650): OTA UPDATE ON STARTING DOWNLOAD
D/FLUTTER OTA(19650): OTA UPDATE DOWNLOAD STARTED 25
D/FLUTTER OTA(19650): OTA UPDATE TRACK DOWNLOAD STARTED 25
D/FLUTTER OTA(19650): OTA UPDATE TRACK DOWNLOAD THREAD STARTED 25
6
D/FLUTTER OTA(19650): OTA UPDATE PENDING
I/chatty  (19650): uid=10126(com.example.reading_app) Thread-4 identical 1 line
2
D/FLUTTER OTA(19650): OTA UPDATE PENDING
I/chatty  (19650): uid=10126(com.example.reading_app) Thread-4 identical 1 line
4
D/FLUTTER OTA(19650): OTA UPDATE PENDING
I/chatty  (19650): uid=10126(com.example.reading_app) Thread-4 identical 6 lines
```

**Comment by MartinHlavna** [comment]
One idea worth double checking:

Are you using https in local network? If not, are you permitting cleartext traffic? AFAIK, download should be FAILED and not PENDING if this is problem, but might be worth a try.

**Comment by ahmadhsalim** [comment]
It's on http on local network and cleartext is permitted. I guess we have to use alternative method.

**Comment by nguhoangictu** [comment]
> It's on http on local network and cleartext is permitted. I guess we have to use alternative method.

I have the same problem with u, did u solve it?

**Comment by MartinHlavna** [comment]
Should be fixed in 4.0.1
